services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -170,12 +172,18 @@ class HankeController(
         }
     }
 
-    private fun Hanke.validateUpdatable(newHanke: Hanke, hankeTunnusPath: String) {
-        if (hankeTunnusPath != newHanke.hankeTunnus) {
+    /**
+     * Verifies that hanke is updatable. Sets generated false as hanke to be updated is not
+     * considered generated anymore.
+     */
+    private fun Hanke.ensureUpdatable(updatedHanke: Hanke, hankeTunnusFromPath: String): Hanke {
+        if (hankeTunnusFromPath != updatedHanke.hankeTunnus) {
             throw HankeArgumentException("Hanketunnus not given or doesn't match the hanke data")
         }
-        if (perustaja != null && perustaja != newHanke.perustaja) {
+        if (perustaja != null && perustaja != updatedHanke.perustaja) {
             throw HankeArgumentException("Updating perustaja not allowed.")
         }
+
+        return updatedHanke.apply { generated = false }
```

Comment:
Same thing here, I think resetting the generated flag should be a part of the business logic. Maybe add it here?

https://github.com/City-of-Helsinki/haitaton-backend/blob/59091c289426cdc6e135f240a131b02b998c3920/services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt#L172-L177